engine/runtime-compiler/src/main/java/org/enso/compiler/phase/ImportResolverForIR.java

```diff
@@ -0,0 +1,163 @@
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.expression.errors.ImportExport;
+import org.enso.compiler.core.ir.module.scope.Export;
+import org.enso.compiler.core.ir.module.scope.Import;
+import org.enso.compiler.data.BindingsMap;
+import org.enso.compiler.data.BindingsMap$ModuleReference$Concrete;
+import org.enso.compiler.data.BindingsMap.ResolvedType;
+import org.enso.editions.LibraryName;
+import org.enso.polyglot.CompilationStage;
+
+import scala.Option;
+import scala.Tuple2;
+import scala.collection.immutable.List;
+import scala.jdk.CollectionConverters;
+
+abstract class ImportResolverForIR extends ImportResolverAlgorithm<
+  Tuple2<Import, Option<BindingsMap.ResolvedImport>>,
+  Module,
+  Import.Module,
+  Export.Module,
+  ResolvedType,
+  CompilerContext.Module
+> {
+  abstract Compiler getCompiler();
+
+  @Override
+  protected final String nameForImport(Import.Module imp) {
+    return imp.name().name();
+  }
+
+  @Override
+  protected final java.util.List<String> partsForImport(Import.Module imp) {
+    return CollectionConverters.SeqHasAsJava(imp.name().parts().map(n -> n.name())).asJava();
+  }
+
+  @Override
+  protected final String nameForExport(Export.Module ex) {
+    return ex.name().name();
+  }
+
+  @Override
+  protected final String nameForType(BindingsMap.ResolvedType e) {
+    return e.qualifiedName().item();
+  }
+
+  @Override
+  protected final java.util.List<Export.Module> exportsFor(Module module, String impName) {
+    java.util.List<Export.Module> exp = CollectionConverters.SeqHasAsJava(module.exports()).asJava().stream().map(e -> switch (e) {
+      case Export.Module ex when ex.name().name().equals(impName) -> ex;
+      case null, default -> null;
+    }).filter(Objects::nonNull).toList();
+    return exp;
+  }
+
+  @Override
+  protected final boolean isAll(Export.Module ex) {
+    return ex.isAll();
+  }
+
+  @Override
+  protected final java.util.List<String> onlyNames(Export.Module ex) {
+    if (ex.onlyNames().isEmpty()) {
+      return null;
+    }
+    var list = CollectionConverters.SeqHasAsJava(ex.onlyNames().get().map(n -> n.name())).asJava();
+    return list;
+  }
+
+  @Override
+  protected final java.util.List<String> hiddenNames(Export.Module ex) {
+    if (ex.hiddenNames().isEmpty()) {
+      return null;
+    }
+    var list = CollectionConverters.SeqHasAsJava(ex.hiddenNames().get().map(n -> n.name())).asJava();
+    return list;
+  }
+
+  @Override
+  protected final java.util.List<BindingsMap.ResolvedType> definedEntities(Import.Module name) {
+    var parts = partsForImport(name);
+    var last = parts.size() - 1;
+    var modName = String.join(".", parts.subList(0, last).stream().toList());
+    var compiler = this.getCompiler();
+    var optionMod = compiler.getModule(modName);
+    if (optionMod.isEmpty()) {
+      return null;
+    }
+    var mod = optionMod.get();
+    compiler.ensureParsed(mod);
+    var b = mod.getBindingsMap();
```

Comment:
```suggestion
    var bindingsMap = mod.getBindingsMap();
```

I think we should prefer to use a bit more meaningful names than just 1-letter ones, it makes the code slightly easier to read, as [suggested by our Java style guide](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md#naming).

Reply:
I have a long history of [using single letter identifiers](https://github.com/apache/netbeans/blob/e074d9d84e4213d016f47b487fc36deb47e773ff/platform/openide.util.lookup/src/org/openide/util/lookup/ProxyLookup.java#L397). However, of course, only in the implementation. Not as a part of the API. E.g. not in signatures that are visible to external users.

In this case the problem is really small, it is a local variable. In the [previous case](https://github.com/apache/netbeans/blob/e074d9d84e4213d016f47b487fc36deb47e773ff/platform/openide.util.lookup/src/org/openide/util/lookup/ProxyLookup.java#L397) it should have been the same (e.g. just a part of internal implementation), but it turned out that the _debugging was slightly complicated_ - my colleagues were not happy to see class called `R` in the debugger.

Reply:
Your suggestion would create broken code. Properly renamed at 859ae93

Reply:
> Your suggestion would create broken code. Properly renamed at [859ae93](https://github.com/enso-org/enso/commit/859ae93b38bc3c18b197520e01467c66e7d9d141)

The suggestion was just for the variable rename, of course all usages had to be updated.

Reply:
Using suggestions in such a case is a bit misleading. They come with "apply" button, but "applying" the suggestion would result in broken code. It was clear to me the code would be broken in this case, but there were situations in the past when I "applied" and only when CI started to fail I realized the suggestion wasn't a fix, but just a sketch.

I suggest to _not use suggestions_ for changes that would create apparently broken code.

Reply:
> Using suggestions in such a case is a bit misleading. They come with "apply" button, but "applying" the suggestion would result in broken code. It was clear to me the code would be broken in this case, but there were situations in the past when I "applied" and only when CI started to fail I realized the suggestion wasn't a fix, but just a sketch.

I see, I did not realize that it would be confusing as I was usually applying these manually most of the time anyway - sorry for the confusion it has caused.

> I suggest to _not use suggestions_ for changes that would create apparently broken code.

Ok, I will try to keep that in mind.